src/javascripts/modules/pillbar.js

```diff
@@ -0,0 +1,53 @@
+export default class Pillbar {
+  constructor(el) {
+    this.el = el
+    this.setVars()
+  }
+
+  setVars(el) {
+    /* Ticking essentially == isRequestingAnimationFrame; I'm leaving
+    it as 'ticking' to follow requestingAnimationFrame convention/docs */
+    this.ticking = false
+    this.lastScrollY = 0
+    this.el.pillbarOffset = this.el.offsetTop
+    this.setupListeners()
```

Comment:
put the `setupListeners` call within the constructor after `setVars`. javascript is will execute each function before moving onto the next one, unless you signify it as `async` so you don't need to worry about it not having the variables set by the time it makes the event listeners (if the listeners rely on variables at all)